src/main/kotlin/domain/strategy/RandomMineBoardGenerateStrategy.kt

```diff
@@ -0,0 +1,40 @@
+package domain.strategy
+
+import domain.Coordinate
+import domain.Field
+import domain.FieldType
+import domain.Fields
+import domain.Land
+import domain.Mine
+import domain.MineCnt
+
+class RandomMineBoardGenerateStrategy : BoardGenerateStrategy {
+    override fun generate(maxCoordinateSize: Int, mineCnt: MineCnt): Fields {
+        val mineCoordinates = getMineCoordinates(maxCoordinateSize, mineCnt)
+        return generateFields(maxCoordinateSize, mineCoordinates)
+    }
+
+    private fun getMineCoordinates(maxCoordinateSize: Int, mineCnt: MineCnt): List<Coordinate> {
+        return (INITIAL_COORDINATE_SIZE until maxCoordinateSize).shuffled()
+            .take(mineCnt.value)
+            .map { number -> Coordinate(number) }
+    }
+
+    private fun generateFields(maxCoordinateSize: Int, mineCoordinate: List<Coordinate>): Fields {
+        return Fields(generateFieldMap(maxCoordinateSize, mineCoordinate))
+    }
+
+    private fun generateFieldMap(maxCoordinateSize: Int, mineCoordinate: List<Coordinate>): Map<Coordinate, Field> {
+        return List(maxCoordinateSize) { number ->
+            val coordinate = Coordinate(number)
+            when (coordinate in mineCoordinate) {
+                true -> coordinate to Mine(coordinate, FieldType.MINE)
+                false -> coordinate to Land(coordinate, FieldType.LAND)
+            }
+        }.toMap()
```

Comment:
```suggestion
        return List(maxCoordinateSize) { number ->
            val coordinate = Coordinate(number)
            when (coordinate in mineCoordinate) {
                true -> Mine(coordinate, FieldType.MINE)
                false -> Land(coordinate, FieldType.LAND)
            }
        }.associateBy { it.coordinate }
```

- 위와 같이 구현할 수도 있어요!
- Fields에서도 key값으로 coordinate를 관리하고, 내부 Field에서도 coordinate을 관리해주는 이유가 있을까요? 같은 데이터를 여러 객체에서 관리하면 동기화에 대한 관리도 해줘야할 것 같아서요.

Reply:
동기화 관리 측면에서 Coordinate를 키로만 관리하고 벨류로 가져가는게 더욱 좋겠네요 👍